src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +36,19 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        List<Integer> modifiedNumbers = IntStream.range(0, numbers.size())
+                .mapToObj(i -> i % 2 == 0 ? numbers.get(i) : numbers.get(i) - 1)
+                .collect(Collectors.toList());
+
+        List<Integer> oddNumbers = modifiedNumbers.stream()
+                .filter(num -> num % 2 != 0)
+                .collect(Collectors.toList());
+
+        if (oddNumbers.isEmpty()) {
+            throw new NoSuchElementException("No odd numbers in the list.");
+        }
+
+        return oddNumbers.stream().mapToInt(Integer::intValue).average().orElseThrow();
```

Comment:
```suggestion
        return IntStream.range(...
```
group everything together, no need to use `if` and `mapToObj` and then `mapToInt`